cmd/handler-utils.go

```diff
@@ -174,6 +186,37 @@ func extractMetadataFromMap(ctx context.Context, v map[string][]string, m map[st
 	return nil
 }
 
+// extractTags extracts tag key (X-Amz-Tagging) and value from given http header. It then
+// - Parses the input format X-Amz-Tagging:"Key1=Value1&Key2=Value2" into a map[string]string
+// with entries in the format X-Amg-Tag-Key1:Value1, X-Amz-Tag-Key2:Value2
+// - Validates the tags
+// - The parsed tags are then merged with input map and returned to caller.
+func extractTags(ctx context.Context, r *http.Request, m map[string]string) (map[string]string, error) {
+	// Check if the metadata has tagging related header
+	val := r.Header.Get(http.CanonicalHeaderKey(xhttp.AmzObjectTagging))
+	if val != "" {
+		tags, err := url.ParseQuery(val)
+		if err != nil {
+			return nil, err
+		}
+		parsedTags := make(map[string]string, len(tags))
+		for key := range tags {
+			if value, ok := tags[key]; ok {
+				// Add the "X-Amz-Tag" prefix
+				parsedTags[xhttp.AmzObjectTagKeyPrefix+key] = strings.Join(value, ",")
```

Comment:
should we have the check for the prefix already? Like, adding the prefix only if it is not found already.. like we do for user-defined metadata? @nitisht

Reply:
Never mind - Clients should have taken care of these things